### DISCUSSION TITLE
incorrect change_type

### DISCUSSION THREAD

**Discussion by grumpyp** [gitpython-developers/GitPython#1561] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/75830792/224020111-d495882b-2484-44ef-ae47-341c38a56221.png)

![image](https://user-images.githubusercontent.com/75830792/224019500-18245897-578a-42a5-98aa-42470f1722b0.png)

shouldn't this be a `A` instead of a `D` as I am not deleting anything?

**Comment by Byron** [comment, marked as answer]
Which way around is right has always been troublesome for me as `diff` doesn't convey the direction into which the diff should go, at least not without prior training.

In `gitoxide`, I use `tree.(get_)changes_to_obtain(other_tree)` which makes clear what needs to be done to turn `tree` into `other_tree`. Without the notion of direction, things can look reversed pretty quickly.

Changing anything in GitPython would be a breaking change and thus can't be done, but either one can add new names to make direction clear, or code has to be adjusted to be `target.diff(index)`.
